### DISCUSSION TITLE
是否有必要添加"节点选择", "管理连接", "查看/过滤 mihomo 日志"的功能?

### DISCUSSION THREAD

**Discussion by JohanChane** [JohanChane/clashtui#33] [Ideas]
我平时用这三个功能时, 都喜欢在网页上操作。如果有人喜欢在 clashtui 上使用这些功能时, 可以添加这些功能。

我现在想到的设计是, 它们三个都有同样的操作方法。

查看并选择节点:
-   'g': 列出所有 proxy-groups
-   'a': 列出所有 proxies (方便过虑)
查看节点:
查看并操作连接:

三个共同的操作 (与 vim 操作类似):
-   选择列表的某一项:
    -   '/': 输入查找的字符串, `n/N`。
    -   'f(ilter)': 输入过滤的字符串。只显示过滤后的结果。
    -   's(elect)': 每个列表选项显示选择它的两个或多个字符。
-   操作: 每个键对应一个操作即可 (select, disconnect, ...)

操作描述:
-   `/`, `f` 操作类似 ranger/joshuto/yazi 等 TUI 文件管理器的操作。
-   `s` 类似于 vimium, folke/flash.nvim 等操作。或者使用 fzf。

三个功能可以使用三个 Tab。只实现一个 TUI widget 即可。

**Comment by Jackhr-arch** [comment]
It's not hard to select proxies or terminate connections, but Mihomo log is different.
Log from restful api is **real time**, that's to say we get logs only after we start a persistent connection(we might need to import a async runtime or do multiprocess for this) to `controller_api/logs`.
Although this can be solved via reading logs from `systemd`, but for those who don't use systemd to manage service, this might not work.

**Reply by JohanChane** [reply]
日志功能放到最后做吧, 等其他功能都完善了再做了。到时有时间静下来再慢慢搞了。
